beacon-chain/rpc/beacon_server.go

```diff
@@ -338,6 +338,76 @@ func (bs *BeaconServer) BlockTree(ctx context.Context, _ *ptypes.Empty) (*pb.Blo
 	}, nil
 }
 
+// BlockTreeBySlots returns the current tree of saved blocks and their votes starting from the justified state.
+func (bs *BeaconServer) BlockTreeBySlots(ctx context.Context, req *pb.TreeBlockSlotRequest) (*pb.BlockTreeResponse, error) {
+	justifiedState, err := bs.beaconDB.JustifiedState()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve justified state: %v", err)
+	}
+	attestationTargets, err := bs.targetsFetcher.AttestationTargets(justifiedState)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve attestation target: %v", err)
+	}
+	justifiedBlock, err := bs.beaconDB.JustifiedBlock()
+	if err != nil {
+		return nil, err
+	}
+	if req == nil {
+		return nil, fmt.Errorf("argument 'TreeBlockSlotRequest' cannot be nil")
+	}
+	if !(req.GetSlotNumberFrom() <= req.GetSlotNumberTo()) {
```

Comment:
can you do this instead `req.SlotNumberFrom` and so on ? Everytime we call the getter function it performs a copy

Reply:
This getter is generated (it's inside the services.pb.go file , generated by the proto script 'update-go-pbs.sh'). The definition of the fields - SlotNumberFrom and SlotNumberTo is done the in the profo files and I couldn't find a way to get to simply be public variables. 
It would be great to get some direction on how to so it.

Reply:
Figured it out. Done.